src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.UnifiedBuild.Tasks/RemoveSectionsFromNugetConfig.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.UnifiedBuild.Tasks
+{
+    /*
+     * This task removes all specified sections from Nuget.Config file.
+     */
+    public class RemoveSectionsFromNugetConfig : Task
+    {
+        [Required]
+        public string NuGetConfigFile { get; set; }
+
+        [Required]
+        public string[] SectionsToRemove { get; set; }
+
+        public override bool Execute()
+        {
+            string xml = File.ReadAllText(NuGetConfigFile);
+            string newLineChars = FileUtilities.DetectNewLineChars(xml);
+            XDocument d = XDocument.Parse(xml);
+            foreach (string sectionName in SectionsToRemove)
+            {
+                d.Root?.Elements(sectionName)?.Remove();
+            }
+
+            using (var w = XmlWriter.Create(NuGetConfigFile, new XmlWriterSettings { NewLineChars = newLineChars, Indent = true }))
```

Comment:
[Personal preference] Consider using statements as they reduce nesting and are less verbose.

```suggestion
            var w = XmlWriter.Create(NuGetConfigFile, new XmlWriterSettings { NewLineChars = newLineChars, Indent = true });
```

Reply:
Most of the code was copied from an existing task. But, I see the value in updating it - will fix.